geom/dcel_overlay.go

```diff
@@ -90,10 +41,17 @@ func (d *doublyConnectedEdgeList) fixVertices() {
 
 func (d *doublyConnectedEdgeList) fixVertex(v *vertexRecord) {
 	// Sort the edges radially.
-	if len(v.incidents) >= 3 {
-		sort.Slice(v.incidents, func(i, j int) bool {
-			ei := v.incidents[i]
-			ej := v.incidents[j]
+	incidents := make([]*halfEdgeRecord, 0, len(v.incidents))
+	for e := range v.incidents {
+		incidents = append(incidents, e)
+	}
+	if len(incidents) >= 3 {
```

Comment:
This question stems from me not fully understanding what this method is doing 😅... why are we specifically checking on len >= 3?

Reply:
This part of the code is "fixing" the `prev` and `next` pointers for each half-edge such that when a chain of `next` (or `prev`) pointers are followed, they form a cycle (similar to a doubly linked list) around a face. The method here is doing that for the edges that are incident to a particular vertex (i.e. the `vertexRecord` argument to the method).

As part of fixing those pointers, we need to sort the edges radially around the vertex. The call to `sort.Slice` sorts the edges by calculating their angle _relative to the positive x-axis_ by using `math.Atan2` (which gives the angle of a line between the origin and a point given the vertical and horizontal displacement of the point from the origin).

_However_, we don't actually care about sorting _relative to the positive x-axis_. All we care about is that the edges are sorted _relative to each other_. Sorting relative to the positive x-axis is just a convenient way to do that. If there are only 1 or 2 incident edges, then the edges are already (trivially) sorted relative to each other. It's only when you get 3 edges that they can potentially be unsorted relative to each other. So essentially this is a performance optimisation, so skip sorting when it's not needed.

I've added some comments to that effect in ef59017d82d9bfbf948bdc9e44e9a18dcd4aaff2.